core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/hdr/InternalHDRPercentilesRanksTests.java

```diff
@@ -61,4 +66,26 @@ protected void assertReduced(InternalHDRPercentileRanks reduced, List<InternalHD
         return InternalHDRPercentileRanks::new;
     }
 
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        return new NamedXContentRegistry(singletonList(
+                new NamedXContentRegistry.Entry(
+                        Aggregation.class,
+                        new ParseField(InternalHDRPercentileRanks.NAME),
+                        (parser, context) -> ParsedHDRPercentileRanks.fromXContent(parser, (String) context))));
```

Comment:
I am wondering if we should have a single final impl for this in the base test class that contains all of the aggs that we can parse. That would be useful later when we will parse nested aggs and bucket aggs.